tests/test_schema.py

```diff
@@ -161,29 +162,37 @@ def remove_param(self, param: str) -> None:
         """
         Remove a parameter from the schema.
 
-        :param param: Name of the parameter to remove.
+        :param param: Name of the parameter to remove
         """
         self.schema["function"]["parameters"]["properties"].pop(param)
-        self.schema["function"]["parameters"]["required"].pop(param, None)
+
+        if (required := self.get_required_parameters()) and param in required:
+            required.remove(param)
 
     def get_param(self, param: str) -> dict:
         """
         Get a parameter dictionary from the schema.
 
-        :param param: Name of the parameter.
-        :return: The parameter dictionary.
+        :param param: Name of the parameter
+        :return: The parameter dictionary
         """
         return self.schema["function"]["parameters"]["properties"][param]
 
     def set_param(self, param, value: dict) -> None:
         """
         Set a parameter dictionary.
 
-        :param param: Name of the parameter.
-        :param value: The new parameter dictionary.
+        :param param: Name of the parameter
+        :param value: The new parameter dictionary
         """
         self.schema["function"]["parameters"]["properties"][param] = value
 
+    def get_required_parameters(self) -> list[str]:
```

Comment:
Should use `Optional` here because technically `get` can return `None` I think.

```suggestion
    def get_required_parameters(self) -> Optional[list[str]]:
```

Reply:
Yep, done in e72566f